{
    "code": "FRA_LINK_BASE_GENERIC", 
    "logs": [
        "Traceback (most recent call last):", 
        "", 
        "  File \"freedomrobotics/link_base.py\", line 578, in freedomrobotics.link_base.LinkUploader.__loop_connection_check_to_freedom_api", 
        "", 
        "Exception: Could not connect to Freedom API after 3 attempts. No Internet connection.", 
        ""
    ], 
    "level": "fatal", 
    "errorMessage": "Unhandled exception in Link `agent_link` freedom api connection check loop: <type 'exceptions.Exception'> Could not connect to Freedom API after 3 attempts. No Internet connection.", 
    "component": "agent", 
    "attributes": {
        "python": {
            "implementation": "CPython", 
            "version": "2.7.17", 
            "build": [
                "default", 
                "Feb 27 2021 15:10:58"
            ]
        }, 
        "thread": {
            "daemon": false, 
            "name": "freedom_agent_link_connection_check_thread"
        }, 
        "geo": {
            "city": "Plano", 
            "zip": "75074", 
            "countryCode": "US", 
            "country": "United States", 
            "region": "TX", 
            "isp": "Charter Communications", 
            "lon": -96.6826, 
            "as": "AS11427 Charter Communications Inc", 
            "query": "70.119.66.130", 
            "org": "Spectrum", 
            "lat": 33.0287, 
            "timezone": "America/Chicago", 
            "regionName": "Texas"
        }, 
        "system": {
            "linux_dist": [
                "Ubuntu", 
                "18.04", 
                "bionic"
            ], 
            "ip": "70.119.66.130", 
            "user": "ubuntu", 
            "arch": "Unknown: aarch64", 
            "service": "systemd", 
            "hostname": "ubuntu", 
            "environment": {
                "LANG": "C.UTF-8", 
                "SHELL": "/bin/bash", 
                "JOURNAL_STREAM": "9:39795", 
                "LOGNAME": "ubuntu", 
                "USER": "ubuntu", 
                "PATH": "/home/ubuntu/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin", 
                "HOME": "/home/ubuntu", 
                "INVOCATION_ID": "8f9b7ce457da47f9a09f516769c7792e"
            }, 
            "platform": "Linux-5.4.0-1035-raspi-aarch64-with-Ubuntu-18.04-bionic", 
            "version": "#38~18.04.1-Ubuntu SMP PREEMPT Wed Apr 21 00:17:21 UTC 2021", 
            "release": "5.4.0-1035-raspi", 
            "docker": false, 
            "os": "Linux", 
            "processor": "aarch64"
        }, 
        "freedom_robotics": {
            "device": "D36F4D9E51F228A3B221D203CB1", 
            "freedom_python_module_version": "0.1.157", 
            "token": "***C390FC37", 
            "account": "A139C78C471C762F34F31F8DD", 
            "code_branch": "release"
        }
    }
}